pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -180,15 +180,22 @@ func (m *kubeGenericRuntimeManager) startContainer(podSandboxID string, podSandb
 			Type: m.runtimeName,
 			ID:   containerID,
 		}
-		msg, handlerErr := m.runner.Run(kubeContainerID, pod, container, container.Lifecycle.PostStart)
-		if handlerErr != nil {
-			m.recordContainerEvent(pod, container, kubeContainerID.ID, v1.EventTypeWarning, events.FailedPostStartHook, msg)
-			if err := m.killContainer(pod, kubeContainerID, container.Name, "FailedPostStartHook", nil); err != nil {
-				klog.Errorf("Failed to kill container %q(id=%q) in pod %q: %v, %v",
-					container.Name, kubeContainerID.String(), format.Pod(pod), ErrPostStartHook, err)
+		postStart := func(async bool) (string, error) {
+			msg, handlerErr := m.runner.Run(kubeContainerID, pod, container, container.Lifecycle.PostStart)
+			if handlerErr != nil {
+				m.recordContainerEvent(pod, container, kubeContainerID.ID, v1.EventTypeWarning, events.FailedPostStartHook, msg)
+				if err := m.killContainer(pod, kubeContainerID, container.Name, "FailedPostStartHook", nil); err != nil {
+					klog.Errorf("Failed to kill container %q(id=%q) in pod %q: %v, %v",
+						container.Name, kubeContainerID.String(), format.Pod(pod), ErrPostStartHook, err)
+				}
+				if async {
+					klog.Errorf("%s: %v", ErrPostStartHook, handlerErr)
+				}
+				return msg, fmt.Errorf("%s: %v", ErrPostStartHook, handlerErr)
 			}
-			return msg, fmt.Errorf("%s: %v", ErrPostStartHook, handlerErr)
```

Comment:
what effect does losing the error propagation have here?

Reply:
this means post start hook failures will no longer be propagated into a call to `startContainerResult.Fail`